frontend/src/context/WalletContext.tsx

```diff
@@ -0,0 +1,48 @@
+import { BigNumber } from "ethers"
+import { createContext, useContext, useEffect, useState } from "react"
+import { getTokenBalance } from "../sdk"
+import { GeyserContext } from "./GeyserContext"
+import Web3Context from "./Web3Context"
+
+export const WalletContext = createContext<{
```

Comment:
what do you think about moving the localstorage logic (for storing the user's selected wallet) in here?

Reply:
I think the localstorage logic is quite tightly coupled with the web3 and bnc-onboard logic. It might be too much extra work to move it here.

Reply:
would it better to combine the wallet selecting logic and the `WalletContext` to an overall `UserContext`?

Reply:
The `Web3Context` might not be able to combine with `WalletContext`: `GeyserContext` needs `Web3Context` as a parent, and `WalletContext` needs `GeyserContext` as a parent. It would be possible to move the `WalletContext` to the `GeyserContext`. The `WalletContext` should really be called "the context of user wallet with respect to the staking token of the currently selected geyser"

Reply:
Ahh I see, thanks for clarifying! Let's leave it as is for now then and revisit it :)